src/jit/codegenxarch.cpp

```diff
@@ -8968,8 +9161,7 @@ void CodeGen::genPutStructArgStk(GenTreePtr treeNode, unsigned baseVarNum)
                     }
 
                     getEmitter()->emitIns_R_AR(ins_Load(memType), emitTypeSize(memType), REG_RCX, REG_RSI, 0);
-                    getEmitter()->emitIns_S_R(ins_Store(memType), emitTypeSize(memType), REG_RCX, baseVarNum,
-                                              ((copiedSlots + putArgStk->gtSlotNum) * TARGET_POINTER_SIZE));
+                    genStoreRegToStackArg(memType, REG_RCX, copiedSlots * TARGET_POINTER_SIZE);
```

Comment:
Same basic answer as above. The `gtSlotNum` goes into computing the initial offset for the beginning of the struct argument, which is set in `m_stkArgOffset`.

---
In reply to: [79509895](https://github.com/dotnet/coreclr/pull/7252#discussion_r79509895) [](ancestors = 79509895)